src/TensorValues/Operations.jl

```diff
@@ -207,6 +207,20 @@ end
   Meta.parse("ThirdOrderTensorValue{$D}($str)")
 end
 
+@generated function dot(a::A,b::B) where {A<:MultiValue{Tuple{D,D}},B<:ThirdOrderTensorValue{D,D,L}} where {D,L}
```

Comment:
Delete previous method (line 196) since it is not needed any more.

Reply:
Sure. I will take a look and make the changes.